src/fmu/dataio/_filedata_provider.py

```diff
@@ -153,13 +155,13 @@ def _get_filestem(self):
             stem = stem.replace("__", "_")
 
         # treat norwegian special letters
+        # BUG(?): What about germen letter like "Ü"?
         stem = stem.replace("æ", "ae")
         stem = stem.replace("ø", "oe")
         return stem.replace("å", "aa")
 
-    def _get_path(self):
+    def _get_path(self) -> tuple[Path, Path | None]:
```

Comment:
```suggestion
    def _get_path(self) -> tuple[Path, Path] | tuple[Path | None]:
```
Maybe this doesn't fix the previous comment about the assert... but could be worth checking